pylint/checkers/similar.py

```diff
@@ -43,15 +69,294 @@
+
+    :param all_couples: collection that has to be cleaned up from successives entries.
+                        The keys are couples of indices that mark the beginning of common entries
+                        in both linesets. The values have two parts. The first one is the couple
+                        of starting and ending line numbers of common successives lines in the first file.
+                        The second part is the same for the second file.
+
+    For example consider the following dict:
+
+    >>> all_couples
+    {(11, 34): ([5, 9], [27, 31]),
+     (23, 79): ([15, 19], [45, 49]),
+     (12, 35): ([6, 10], [28, 32])}
+
+    There are two successives keys (11, 34) and (12, 35).
+    It means there are two consecutive similar chunks of lines in both files.
+    Thus remove last entry and update the last line numbers in the first entry
+
+    >>> remove_successives(all_couples)
+    >>> all_couples
+    {(11, 34): ([5, 10], [27, 32]),
+     (23, 79): ([15, 19], [45, 49])}
+    """
+    couple: LineSetStartCouple
+    for couple in tuple(all_couples.keys()):
+        to_remove = []
+        test = couple.increment(Index(1))
+        while test in all_couples:
+            all_couples[couple].first_file.end = all_couples[test].first_file.end
+            all_couples[couple].second_file.end = all_couples[test].second_file.end
+            all_couples[couple].effective_cmn_lines_nb += 1
+            to_remove.append(test)
+            test = test.increment(Index(1))
+
+        for target in to_remove:
+            try:
+                all_couples.pop(target)
+            except KeyError:
+                pass
+
+
+def filter_noncode_lines(
+    ls_1: "LineSet",
+    stindex_1: Index,
+    ls_2: "LineSet",
+    stindex_2: Index,
+    common_lines_nb: int,
+) -> int:
+    """
+    Return the effective number of common lines between lineset1 and lineset2 filtered from non code lines, that is to say the number of
+    common successive stripped lines except those that do not contain code (for example a ligne with only an
+    ending parathensis)
+
+    :param ls_1: first lineset
+    :param stindex_1: first lineset starting index
+    :param ls_2: second lineset
+    :param stindex_2: second lineset starting index
+    :param common_lines_nb: number of common successive stripped lines before being filtered from non code lines
+    :return: the number of common successives stripped lines that contain code
+    """
+    stripped_l1 = [
+        lspecif.text
+        for lspecif in ls_1.stripped_lines[stindex_1 : stindex_1 + common_lines_nb]
+        if REGEX_FOR_LINES_WITH_CONTENT.match(lspecif.text)
+    ]
+    stripped_l2 = [
+        lspecif.text
+        for lspecif in ls_2.stripped_lines[stindex_2 : stindex_2 + common_lines_nb]
+        if REGEX_FOR_LINES_WITH_CONTENT.match(lspecif.text)
+    ]
+    return sum(sline_1 == sline_2 for sline_1, sline_2 in zip(stripped_l1, stripped_l2))
+
+
+class CommonalitiesT(NamedTuple):
```

Comment:
```suggestion
class Commonality(NamedTuple):
```
This should probably be singular (?), as well as without the `T`.

Reply:
Totally right!